src/core/public/application/application_service.tsx

```diff
@@ -311,6 +318,30 @@ export class ApplicationService {
         );
       },
     };
+
+    // do not add the global handler in legacy mode as it would perform a full refresh anyway
+    // also, there is a trick with 'multiple' legacy apps such as kibana:XXX we don't want to
+    // handle to avoid adding complexity to the handler's logic.
+    if (!this.legacyMode) {
+      this.linkClickDelegateHandler = e => {
+        const link = e.currentTarget as HTMLAnchorElement;
+        if (
+          link.href && // ignore links with empty hrefs
+          !selfOrParentHasClass(link, 'disableCoreNavigation') && // ignore explicitly ignored links
+          e.button === 0 && // ignore everything but left clicks
+          !isModifiedEvent(e) // ignore clicks with modifier keys
+        ) {
+          const appInfo = parseAppUrl(link.href, http.basePath, this.apps);
+          if (appInfo && appInfo.app !== this.currentAppId$.value) {
+            e.preventDefault();
+            startContract.navigateToApp(appInfo.app, { path: appInfo.path });
+          }
+        }
+      };
+      on('click', 'a', this.linkClickDelegateHandler);
```

Comment:
Wonder what this lib improves in our case comparing to just listening to `document.body` or `window`?

Reply:
It properly handle events bubbling from child elements.

I.E
```html
<a href="...">
  <span>I'm the content of the link</span>
<a/>
```

Without the lib, adding a global listener on body for `a` and clicking on the link (and effectively on the inner `span`) would not be catched properly by a 'plain' listener.

The bubbling / currentTarget is not **that** complicated, but I don't think recoding it is really worth it there.